### PR TITLE
Adding flags to disable notifications & surveys to avoid unneeded decide requests

### PR DESCRIPTION
**Pull request by gmerinojimenez**
Hello,

in our application we use the Mixpanel track to send usage statistics, but we don't use the notification/surveys features. We've seen that we're sending so many unnecessary requests to _decide_, just to get always the same empty response. We thought that it would be interesting to be able to configure the _decide_ behaviour to just not perform those requests when we are sure that we will not use them.

Thanks in advance.

### CODE REVIEW

**Comment by dpastor** [comment]
Nice job @gmerinojimenez !

**Comment by alexmiragall** [comment]
Really useful!

**Comment by rafaelbartolome** [comment]
Great!

**Comment by PradeMismo** [comment]
Simple and Useful.

**Comment by yinfeiru** [comment]
@gmerinojimenez Thanks for the PR! It looks really good! I've created a PR at #317 and will track the status there.

**Comment by yinfeiru** [comment]
I've had a discussion with @ryanseams and @patedit at #317 and we believe this PR needs a refactor. @gmerinojimenez can you please check out the comments at #317 and make the change accordingly? Basically, we want to have a single flag called something like `DisableDecideChecker`.

I'll go ahead and re-open this. (sorry for closing this one and opened a new PR. I was trying to make some quick changes myself in that PR but it turns out that creating a new PR is totally a bad idea!)

File: src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java [hunk 1]
```diff
@@ -230,7 +230,10 @@
         }
         mDecideMessages.setDistinctId(decideId);
         mMessages = getAnalyticsMessages();
-        mMessages.installDecideCheck(mDecideMessages);
+
+		if (!mConfig.getDisableSurveys() || !mConfig.getDisableNotifications()) {
```
Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java]: Can you please make sure each tab takes 4 spaces?

**Comment by gmerinojimenez** [comment]
Hello @yinfeiru,

thanks for your time, I've updated the pull request with the suggestions from #317. Please, don't hesitate to write me if you want something else.

**Comment by yinfeiru** [comment]
@gmerinojimenez Cool! It looks good. What you think? @patedit

**Comment by patedit** [comment]
This looks good :+1: Let's merge it!

**Comment by yinfeiru** [comment]
merged!
@gmerinojimenez thank you!

**Pull request merged**